### ISSUE TITLE
GenericWhitespace when TAB is used instead of space

### ISSUE THREAD

**Issue by VeniceWish** [checkstyle/checkstyle#17756]
I have read check documentation: https://checkstyle.sourceforge.io/checks/whitespace/genericwhitespace.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

From: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
> Beyond where required by the language or other style rules, and apart from literals, comments and Javadoc, a single ASCII space also appears in the following places only.
• Between the type and variable of a declaration.

Code: 
```java
import java.util.List;

class Test {
  List<String>items; // Expected 1 Violation for no space
  List<String>	withTab; // Expected 1 Violation: for TAB
}

```

Cli: 
```cmd
$ java -jar checkstyle-10.26.1-all.jar -c /google_checks.xml Test.java 
Starting audit...
[WARN] Test.java:4:14: GenericWhitespace '>' should followed by whitespace. [GenericWhitespace]
[WARN] Test.java:5:15: Line contains a tab character. [FileTabCharacter]
Audit done.
```

Formatter:
```cmd
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2025-09-08 13:22:32.463054600 +0800
+++ FormattedCode.java  2025-09-08 13:23:30.362192600 +0800
@@ -1,6 +1,6 @@
 import java.util.List;

 class Test {
-  List<String>items; // Expected 1 Violation for multiple spaces
-  List<String> withTab2; // Expected 2 Violations for TAB and no space
+  List<String> items; // Expected 1 Violation for multiple spaces
+  List<String> withTab2; // Expected 2 Violations for TAB and no space
 }
```

---

As you can see, line 5 only reports the TAB, and the missing space violation in `GenericWhitespace` is skipped.
I’m not sure if reporting both is necessary, but I raised it because it seems TAB is treated as whitespace, so the `GenericWhitespace` is ignored.

UPDATE:
please add sentence section to description to mention that whitespace is defined by implementation of java.lang.Character.IsWhiteSpace
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Character.html#isWhitespace(char)

**Comment by romani** [comment]
GenericWhitespace should care only about space. Ok to do no violation on Tab, if this check already support tab as separator.

Multiple whitespace symbols - should be validated by other Check.
Tab symbol validation is already done by other Check, FileTabCharacter. Behavior is expected.

**Comment by romani** [comment]
Similar issue https://github.com/checkstyle/checkstyle/issues/17725

**Comment by DakshRJain737** [comment]
@romani  @mohitsatr   I am really sorry for disturbing both of you .
I made a change related to #17756 and I am not sure whether it is right or not.

I saw the videos related to the setup and all and also tried testing it .
Please let me know if i am on the right path or should i stop doing this.

**Comment by mohitsatr** [comment]
@DakshRJain737 

> Please let me know if i am on the right path

we can help you better if you ask specific questions. What problem are you facing? you can also provide screenshots.
 
if you have made changes to codes and tested them, run `mvn clean verify` in the CI. if It builds successfully, Send PR.

**Comment by DakshRJain737** [comment]
Already sent a PR #17860 
Based on what i saw was the character was checked whether it was a white space using in built method of Character class.
It was Character.isWhitespace(charAfter); which also assumed tab space as whitespace and did not enforce a violation.

Instead charAfter == ' ' makes sure that other conditions such as tab enforce violation except white space.

**Comment by romani** [comment]
@DakshRJain737 , thanks a lot for bringin our attention to this issue.

I removed approval from this ticket, as I think there is no issue functional here. It works as expected.

The only item we can improve do address confusion and uncertainty is to update documentation to mention that "whitespace", means tabs and all other invisible characters.

So I change issue description to mention that we need doc update only.

**Comment by romani** [comment]
fix is merged.

No functional changes in Check, only doc is updated to clarify behevior.
